### PR TITLE
Set up CORS and OAuth2 Resource Server for Spring API

### PR DESCRIPTION
**Pull request by olehio-p**
- Configured allowed origins for CORS to permit frontend URLs (production and development environments).
- Implemented OAuth2 resource server with JWT validation for secured API endpoints.
- Updated security filter chain to enable OAuth2 authentication in production and permit all requests in development.

### CODE REVIEW

File: backend/src/main/resources/application-prod.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      on-profile: prod
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          issuer-uri: ${AUTH0_ISSUER}}
+
+auth0:
+  audience: ${AUTH0_AUDIENCE} # https://tokoro-back-133c2efe4dcf.herokuapp.com
+  issuer: ${AUTH0_ISSUER} # https://dev-dngu4n3glnkuwbbp.us.auth0.com/
```
Comment (rojikaru) [inline backend/src/main/resources/application-prod.yml]: move auth0 properties out of public files into secrets.properties (and erase comments)

File: backend/src/main/resources/application-prod.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+spring:
+  config:
+    activate:
+      on-profile: prod
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          issuer-uri: ${AUTH0_ISSUER}}
```
Comment (rojikaru) [inline backend/src/main/resources/application-prod.yml]: remove redundant bracket

File: backend/src/main/resources/application-prod.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+spring:
+  config:
+    activate:
+      on-profile: prod
```
Comment (rojikaru) [inline backend/src/main/resources/application-prod.yml]: why specify activate on-profile if this file will be loaded on spring.profiles.active=prod anyway?

File: backend/src/main/java/com/tokorokoshi/tokoro/security/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    private boolean isProduction() {
+        String[] activeProfiles = env.getActiveProfiles();
+        for (String profile : activeProfiles) {
+            if (profile.equals("prod")) {
+                return true;
+            }
+        }
+        return false;
```
Comment (rojikaru) [inline backend/src/main/java/com/tokorokoshi/tokoro/security/SecurityConfig.java]: rewrite to return stream anymatch result

**Review by rojikaru — CHANGES_REQUESTED** [review]

**Comment by rojikaru** [comment]
Reformat chained calls with align enabled
https://blog.jetbrains.com/wp-content/uploads/2020/06/adjust-code-styling-aligned-dots.gif

File: backend/src/main/resources/application-prod.yml (hunk 2 above)
Comment (olehio-p) [reply backend/src/main/resources/application-prod.yml]: Done

File: backend/src/main/resources/application-prod.yml (hunk 3 above)
Comment (olehio-p) [reply backend/src/main/resources/application-prod.yml]: Done

File: backend/src/main/java/com/tokorokoshi/tokoro/security/SecurityConfig.java (hunk 4 above)
Comment (olehio-p) [reply backend/src/main/java/com/tokorokoshi/tokoro/security/SecurityConfig.java]: Done

File: backend/src/main/resources/application-prod.yml (hunk 1 above)
Comment (olehio-p) [reply backend/src/main/resources/application-prod.yml]: Done

**Review by rojikaru — APPROVED** [review]
Next time describe changes in commit messages instead of writing 'fix issues'

**Review by rojikaru — APPROVED** [review]

**Pull request merged**